src/field_theory/separable.lean

```diff
@@ -482,3 +501,48 @@ end
 the minimal polynomial of every `x : K` is separable. -/
 @[class] def is_separable (F K : Sort*) [field F] [field K] [algebra F K] : Prop :=
 ∀ x : K, ∃ H : is_integral F x, (minimal_polynomial H).separable
+
+lemma is_separable_top (F K E : Type*) [field F] [field K] [field E] [algebra F K] [algebra F E]
+  [algebra K E] [is_scalar_tower F K E] (h : is_separable F E) : is_separable K E :=
+begin
+  intro x,
+  cases h x with hx hs,
+  have hx' : is_integral K x := is_integral_of_is_scalar_tower x hx,
+  use hx',
+  have key : (minimal_polynomial hx') ∣ (minimal_polynomial hx).map(algebra_map F K),
+  { apply minimal_polynomial.dvd,
+    rw [aeval_def,eval₂_map, ←is_scalar_tower.algebra_map_eq F K E],
+    apply minimal_polynomial.aeval, },
+  cases key with q hq,
+  apply polynomial.separable.of_mul_left,
+  rw ←hq,
+  exact polynomial.separable.map hs,
+end
+
+lemma is_separable_bottom (F K E : Type*) [field F] [field K] [field E] [algebra F K] [algebra F E]
+  [algebra K E] [is_scalar_tower F K E] (h : is_separable F E) : is_separable F K :=
+begin
+  intro x,
+  have main : ∀ p : polynomial F, algebra_map K E (aeval x p) = aeval (algebra_map K E x) p,
+  { intro p,
+    rw [aeval_def,aeval_def,eval₂_eq_eval_map (algebra_map F E),
+    is_scalar_tower.algebra_map_eq F K E,←polynomial.map_map,
+    ←eval₂_eq_eval_map,eval₂_hom,eval₂_eq_eval_map], },
+  have swap : ∀ p : polynomial F, aeval (algebra_map K E x) p = 0 → aeval x p = 0,
+  { intros p hp,
+    rw [←main p,←(algebra_map K E).map_zero] at hp,
+    exact ring_hom.injective (algebra_map K E) hp, },
+  cases h (algebra_map K E x) with hx hs,
+  have hx' : is_integral F x,
+  { cases hx with p hp,
+    use p,
+    exact ⟨hp.1,swap p hp.2⟩, },
+  use hx',
```

Comment:
You can turn this into a lemma in `ring_theory.integral_closure`, of the form `[field S] [is_scalar_tower R S T] : is_integral R T -> is_integral R S`. (I think those are sufficient assumptions.)

Reply:
I've just done this. It turns out that you also need to assume `[nontrivial T]` and that this follows pretty directly from another lemma already in `ring_theory.integral_closure` which instead uses the assumption that `algebra_map S T` is injective. I also added a couple lemmas in `ring_theory.algebra_tower` to make this proof (the one in `field_theory.separable`) a bit more straightforward.